pulsar-client-kafka-compat/pulsar-client-kafka/src/main/java/org/apache/kafka/clients/consumer/PulsarKafkaConsumer.java

```diff
@@ -192,17 +192,23 @@ public void subscribe(Collection<String> topics) {
                 if (partitionMetadata.partitions > 1) {
                     // Subscribe to each partition
                     conf.setConsumerName(ConsumerName.generateRandomName());
+                    List<CompletableFuture<org.apache.pulsar.client.api.Consumer>> futures = new ArrayList<>();
                     for (int i = 0; i < partitionMetadata.partitions; i++) {
                         String partitionName = DestinationName.get(topic).getPartition(i).toString();
-                        org.apache.pulsar.client.api.Consumer consumer = client.subscribe(partitionName, groupId, conf);
-                        consumers.put(new TopicPartition(topic, i), consumer);
+                        futures.add(client.subscribeAsync(partitionName, groupId, conf));
                     }
+
+                    // Wait for all consumers to be ready
+                    for (int i = 0; i < partitionMetadata.partitions; i++) {
+                        consumers.putIfAbsent(new TopicPartition(topic, i), futures.get(i).get());
```

Comment:
what if one of the partitioned consumer fails to subscribe? 
eg: out of c1,c2,c3 -> c2 fails so, do we have to close c1 and c3? and should we remove c1 from `consumers` because we already added c1?